### PR TITLE
#2066 doctest update layers

### PR DESCRIPTION
**Pull request by nataliyah123**
part of #2066

### CODE REVIEW

File: tensorflow_addons/layers/spectral_normalization.py [hunk 1]
```diff
@@ -26,17 +26,21 @@ class SpectralNormalization(tf.keras.layers.Wrapper):
...
-        tf.keras.layers.Conv2D(2, 2, activation="relu"),
-        input_shape=(32, 32, 3))(x)
-    net = SpectralNormalization(
-        tf.keras.layers.Conv2D(16, 5, activation="relu"))(net)
-    net = SpectralNormalization(
-        tf.keras.layers.Dense(120, activation="relu"))(net)
-    net = SpectralNormalization(
-        tf.keras.layers.Dense(n_classes))(net)
-    ```
+    >>> x = tf.constant( [[ 177,  164,  169,  165,  133,  134,  168,  174,  160,  165], [154,  146,  100,   54,   48,   19,    8,   6,
```
Comment (charlielito) [inline tensorflow_addons/layers/spectral_normalization.py:29]: Why the change to `>>>` for examples?

File: tensorflow_addons/layers/spectral_normalization.py [hunk 2]
```diff
@@ -26,17 +26,21 @@ class SpectralNormalization(tf.keras.layers.Wrapper):
...
+    ... 4,  148], [162,  152,   94,  200,  177,   73,   27,    1,  145,  153], [186,   56,  177,  202,  176,  132,   51,
+    ... 13,   43,  152], [186,   17,  151,  162,  119,  106,   70,   11,   25,  100], [146,   12,  154,  141,  140,  146,
+    ... 92,   21,   16,  101],  [93,    9,  192,  150,  148,  148,   42,   26,   19,   92], [ 91,   13,  125,  129,  114,
+    ... 127,   42,   27,    6,   86], [113,   43,    9,  190,  125,   58,   38,   26,   12,   99], [134,   29,   19,   17,
+    ... 72,   42,   61,   34,   12,   83]],dtype = tf.int32)
```
Comment (charlielito) [inline tensorflow_addons/layers/spectral_normalization.py:34]: The shape of `x` does not work for this example. Also I think if we are going to specify `x`, better use something like `tf.random.normal(shape=(1,10,10,1))` instead of defining explicitly this tensor

File: tensorflow_addons/layers/spectral_normalization.py (hunk 1 above)
Comment (WindQAQ) [reply tensorflow_addons/layers/spectral_normalization.py]: This is for https://github.com/tensorflow/addons/issues/2066.

File: tensorflow_addons/layers/multihead_attention.py [hunk 3]
```diff
@@ -26,34 +26,26 @@ class MultiHeadAttention(tf.keras.layers.Layer):
     [Attention Is All You Need](https://arxiv.org/abs/1706.03762) which takes
     in the tensors `query`, `key`, and `value`, and returns the dot-product attention
     between them:
+    >>> mha = MultiHeadAttention(head_size=128, num_heads=12)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:29]: Add an empty line above this line.

File: tensorflow_addons/layers/multihead_attention.py [hunk 4]
```diff
@@ -26,34 +26,26 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
+    >>> query = tf.random.uniform((3, 5, 4)) # (batch_size, query_elements, query_depth)
+    >>> key = tf.random.uniform((3, 6, 5)) # (batch_size, key_elements, key_depth)
+    >>> value = tf.random.uniform((3, 6, 6)) # (batch_size, key_elements, value_depth)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:33]: remove the trivial empty line.

File: tensorflow_addons/layers/multihead_attention.py [hunk 5]
```diff
@@ -26,34 +26,26 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
-    ```python
-    mha = MultiHeadAttention(head_size=128, num_heads=12)
-
-    query = tf.random.uniform((32, 20, 200)) # (batch_size, query_elements, query_depth)
-    key = tf.random.uniform((32, 15, 300)) # (batch_size, key_elements, key_depth)
-    value = tf.random.uniform((32, 15, 400)) # (batch_size, key_elements, value_depth)
-
-    attention = mha([query, key, value]) # (batch_size, query_elements, value_depth)
-    ```
+    >>> attention = mha([query, key, value]) # (batch_size, query_elements, value_depth)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:34]: can we at print the shape of it? say
```
>>> attention.shape
```

File: tensorflow_addons/layers/multihead_attention.py [hunk 6]
```diff
@@ -26,34 +26,26 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
-
-    query = tf.random.uniform((32, 20, 200)) # (batch_size, query_elements, query_depth)
-    key = tf.random.uniform((32, 15, 300)) # (batch_size, key_elements, key_depth)
-
-    attention = mha([query, key]) # (batch_size, query_elements, key_depth)
-    ```
+    >>> mha = MultiHeadAttention(head_size=128, num_heads=12)
+    >>> query = tf.random.uniform((3, 5, 5)) # (batch_size, query_elements, query_depth)
+    >>> key = tf.random.uniform((3, 6, 10)) # (batch_size, key_elements, key_depth)
+    >>> attention = mha([query, key]) # (batch_size, query_elements, key_depth)
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:41]: ditto.

File: tensorflow_addons/layers/multihead_attention.py [hunk 7]
```diff
@@ -26,34 +26,26 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
 
     Arguments:
         head_size: int, dimensionality of the `query`, `key` and `value` tensors
             after the linear transformation.
         num_heads: int, number of attention heads.
         output_size: int, dimensionality of the output space, if `None` then the
-            input dimension of `value` or `key` will be used,
+        input dimension of `value` or `key` will be used,
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:48]: revert unrelated change.

File: tensorflow_addons/layers/polynomial.py [hunk 8]
```diff
@@ -40,15 +40,14 @@ class PolynomialCrossing(tf.keras.layers.Layer):
 
     Example:
 
-    ```python
+    python
```
Comment (WindQAQ) [inline tensorflow_addons/layers/polynomial.py:43]: remove this line.

File: tensorflow_addons/layers/polynomial.py [hunk 9]
```diff
@@ -40,15 +40,14 @@ class PolynomialCrossing(tf.keras.layers.Layer):
...
     # after embedding layer in a functional model:
```
Comment (WindQAQ) [inline tensorflow_addons/layers/polynomial.py:44]: you can also prefix this with `>>>`

File: tensorflow_addons/layers/spectral_normalization.py (hunk 2 above)
Comment (WindQAQ) [reply tensorflow_addons/layers/spectral_normalization.py]: +1

File: tensorflow_addons/layers/spectral_normalization.py (hunk 1 above)
Comment (nataliyah123) [reply tensorflow_addons/layers/spectral_normalization.py]: @charlielito.Thanks for the review. >>> is used to change the docstring to doctest. Below is the issue
https://github.com/tensorflow/addons/issues/2066

File: tensorflow_addons/layers/spectral_normalization.py (hunk 2 above)
Comment (nataliyah123) [reply tensorflow_addons/layers/spectral_normalization.py]: @charlielito. Thanks a lot for the suggestion it worked and saved me a lot of time. Can you please suggest what value should n_classes get.

File: tensorflow_addons/layers/multihead_attention.py (hunk 3 above)
Comment (nataliyah123) [reply tensorflow_addons/layers/multihead_attention.py:30]: Thanks for the review. made the changes you asked for.

File: tensorflow_addons/layers/multihead_attention.py (hunk 6 above)
Comment (nataliyah123) [reply tensorflow_addons/layers/multihead_attention.py:41]: The author included it. should I remove it or let it be like this

**Comment by WindQAQ** [comment]
Can you fix the errors?

```
#23 287.3 =================================== FAILURES ===================================
#23 287.3 _______ [doctest] tensorflow_addons.layers.polynomial.PolynomialCrossing _______
#23 287.3 [gw1] linux -- Python 3.5.9 /usr/local/bin/python
#23 287.3 037     and diag_scale increases the diagonal of W to improve training stability (especially for the low rank case).
#23 287.3 038 
#23 287.3 039     See [Deep & Cross Network for Ad Click Predictions](https://arxiv.org/pdf/1708.05123.pdf).
#23 287.3 040 
#23 287.3 041     Example:
#23 287.3 042 
#23 287.3 043     # after embedding layer in a functional model:
#23 287.3 044     >>> input = tf.keras.Input(shape=(None,), name='index', dtype=tf.int64)
#23 287.3 045     >>> x0 = tf.keras.layers.Embedding(input_dim=32, output_dim=6)
#23 287.3 046     >>> x1 = PolynomialCrossing(projection_dim=None)((x0, x0))
#23 287.3 UNEXPECTED EXCEPTION: ValueError('Input shapes must be a tuple or list of size 2, got None',)
#23 287.3 Traceback (most recent call last):
#23 287.3   File "/usr/local/lib/python3.5/doctest.py", line 1321, in __run
#23 287.3     compileflags, 1), test.globs)
#23 287.3   File "<doctest tensorflow_addons.layers.polynomial.PolynomialCrossing[2]>", line 1, in <module>
#23 287.3   File "/usr/local/lib/python3.5/site-packages/tensorflow/python/keras/engine/base_layer.py", line 982, in __call__
#23 287.3     self._maybe_build(inputs)
#23 287.3   File "/usr/local/lib/python3.5/site-packages/tensorflow/python/keras/engine/base_layer.py", line 2643, in _maybe_build
#23 287.3     self.build(input_shapes)  # pylint:disable=not-callable
#23 287.3   File "/addons/tensorflow_addons/layers/polynomial.py", line 103, in build
#23 287.3     "got {}".format(input_shape)
#23 287.3 ValueError: Input shapes must be a tuple or list of size 2, got None
#23 287.3 /addons/tensorflow_addons/layers/polynomial.py:46: UnexpectedException
#23 287.3 _ [doctest] tensorflow_addons.layers.spectral_normalization.SpectralNormalization _
#23 287.3 [gw1] linux -- Python 3.5.9 /usr/local/bin/python
#23 287.3 028 
#23 287.3 029     >>> x = tf.random.normal(shape=(1,10,10,1))
#23 287.3 030     >>> net = SpectralNormalization(
#23 287.3 031     ... tf.keras.layers.Conv2D(2, 2, activation="relu"),
#23 287.3 032     ... input_shape=(10, 10))(x)
#23 287.3 033     >>> net = SpectralNormalization(
#23 287.3 034     ... tf.keras.layers.Conv2D(8, 5, activation="relu"))(net)
#23 287.3 035     >>> net = SpectralNormalization(
#23 287.3 036     ... tf.keras.layers.Dense(120, activation="relu"))(net)
#23 287.3 037     >>> net = SpectralNormalization(
#23 287.3 UNEXPECTED EXCEPTION: NameError("name 'n_classes' is not defined",)
#23 287.3 Traceback (most recent call last):
#23 287.3   File "/usr/local/lib/python3.5/doctest.py", line 1321, in __run
#23 287.3     compileflags, 1), test.globs)
#23 287.3   File "<doctest tensorflow_addons.layers.spectral_normalization.SpectralNormalization[4]>", line 2, in <module>
#23 287.3 NameError: name 'n_classes' is not defined
#23 287.3 /addons/tensorflow_addons/layers/spectral_normalization.py:37: UnexpectedException
#23 287.3 _______ [doctest] tensorflow_addons.layers.wrappers.WeightNormalization ________
#23 287.3 [gw1] linux -- Python 3.5.9 /usr/local/bin/python
#23 287.3 024 Performs weight normalization.
#23 287.3 025 
#23 287.3 026     This wrapper reparameterizes a layer by decoupling the weight's
#23 287.3 027     magnitude and direction.
#23 287.3 028     This speeds up convergence by improving the
#23 287.3 029     conditioning of the optimization problem.
#23 287.3 030 
#23 287.3 031     See [Weight Normalization: A Simple Reparameterization to Accelerate Training of Deep Neural Networks](https://arxiv.org/abs/1602.07868).
#23 287.3 032     >>> net = WeightNormalization(
#23 287.3 UNEXPECTED EXCEPTION: NameError("name 'x' is not defined",)
#23 287.3 Traceback (most recent call last):
#23 287.3   File "/usr/local/lib/python3.5/doctest.py", line 1321, in __run
#23 287.3     compileflags, 1), test.globs)
#23 287.3   File "<doctest tensorflow_addons.layers.wrappers.WeightNormalization[0]>", line 4, in <module>
#23 287.3 NameError: name 'x' is not defined
#23 287.3 /addons/tensorflow_addons/layers/wrappers.py:32: UnexpectedException
```

**Comment by nataliyah123** [comment]
fixed the polynomial crossing doctest.

File: tensorflow_addons/layers/multihead_attention.py (hunk 6 above)
Comment (WindQAQ) [reply tensorflow_addons/layers/multihead_attention.py:41]: @nataliyah123 can you follow the comments here?

Comment (nataliyah123) [reply tensorflow_addons/layers/multihead_attention.py:41]: @WindQAQ sorry it's not clear to me what you are asking. do you mean that two sets of code in docstring are similar in the file multihead_attention? I am referring to lines 30 to 41 in my branch

Comment (WindQAQ) [reply tensorflow_addons/layers/multihead_attention.py:41]: I mean can you print `attention.shape` as well like
```
>>> attention.shape
```

Comment (nataliyah123) [reply tensorflow_addons/layers/multihead_attention.py:41]: I haven't tried it. let me try and then I will get back to you,

Comment (nataliyah123) [reply tensorflow_addons/layers/multihead_attention.py:42]: @WindQAQ I am getting the answer as this TensorShape([3, 5, 6]) but like my other pr (metrics). it is failing the build.

**Comment by nataliyah123** [comment]
@gabrieldemarmiesse. Can you please add the cla:yes label or tell me what to do to clear the CLA/google check. I had to close one of my previous pr and open a new one just because of this. Things I have done.

-  I had pulled the master first then checked out to the new branch and pulled the branch associated with this pr on my local system. still facing the problem.

**Comment by nataliyah123** [comment]
@googlebot I fixed it.

**Comment by WindQAQ** [comment]
@nataliyah123 Can you reopen a new PR for it? Seems that the branch is messed up. FYI, the way I use to pull the latest master is

```bash
git remote add upstream https://github.com/tensorflow/addons.git
git checkout your-branch
git pull upstream master
```

**Comment by nataliyah123** [comment]
> @nataliyah123 Can you reopen a new PR for it? Seems that the branch is messed up. FYI, the way I use to pull the latest master is
> 
> ```shell
> git remote add upstream https://github.com/tensorflow/addons.git
> git checkout your-branch
> git pull upstream master
> ```

@WindQAQ I have the upstream as my remote too. The reason I pulled my own branch from origin was I had made the changes online(github) which I wanted to pull. what you have described is for a brand new branch without anything to be pulled from origin. Google bot does not like new email addresses which are not registered in the form and removes the cla.yes label and the workflow I described here is because of the comment made by gabrieldemarmiesse [here](https://github.com/tensorflow/addons/pull/1129)

**Comment by nataliyah123** [comment]
opening a new branch as requested.
